docs/user-guide/guides/pinecone-v7.md

```diff
@@ -0,0 +1,114 @@
+## Pinecone v7 integration
+
+This guide shows how to use Semantic Router with the Pinecone Python SDK v7+, including cloud vs local setup, shared-index reuse, and namespaces for isolation.
+
+### Install
+
+```bash
+pip install "semantic-router[pinecone]"
+```
+
+### Environment variables
+
+- `PINECONE_API_KEY` (required): Your Pinecone API key
+- `PINECONE_API_BASE_URL` (optional):
+  - Cloud: `https://api.pinecone.io` (default if a real API key is set)
+  - Local emulator: `http://localhost:5080` or `http://pinecone:5080`
+- `PINECONE_INDEX_NAME` (recommended on cloud): Name of an existing index to reuse
+
+Why set `PINECONE_INDEX_NAME`? Pinecone serverless has per-project index limits. Reusing a shared index avoids 403 quota errors. Semantic Router will automatically isolate data using namespaces.
+
+### Basic usage (cloud)
+
+```python
+import os
+from semantic_router.encoders import OpenAIEncoder
+from semantic_router.index.pinecone import PineconeIndex
+from semantic_router.route import Route
+from semantic_router.routers import SemanticRouter
+
+# Required
+os.environ["PINECONE_API_KEY"] = "<your-key>"
+
+# Cloud endpoint (optional, defaults to cloud when a real key is present)
+os.environ["PINECONE_API_BASE_URL"] = "https://api.pinecone.io"
```

Comment:
can we drop this? We would only usually change this for tests so it's confusing having it here imo